pyproject.toml

```diff
@@ -10,4 +10,102 @@ requires = [
 ]
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "cheroot"
+description = "Highly-optimized, pure-python HTTP server"
+authors = [
+    { name = "CherryPy Team", email = "team@cherrypy.dev" },
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Environment :: Web Environment",
+    "Intended Audience :: Developers",
+    "Operating System :: OS Independent",
+    "Framework :: CherryPy",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: Implementation",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Programming Language :: Python :: Implementation :: Jython",
+    "Programming Language :: Python :: Implementation :: PyPy",
+    "Topic :: Internet :: WWW/HTTP",
+    "Topic :: Internet :: WWW/HTTP :: HTTP Servers",
+    "Topic :: Internet :: WWW/HTTP :: WSGI",
+    "Topic :: Internet :: WWW/HTTP :: WSGI :: Server",
+    "Typing :: Typed",
+]
+keywords = [
+    "http",
+    "server",
+    "ssl",
+    "wsgi",
+]
+requires-python = ">=3.8"
+dependencies = [
+    "importlib_metadata; python_version < '3.8'",
+    "more_itertools >= 2.6",
+    "jaraco.functools",
+]
+dynamic = ["version"]
+
+[project.urls]
+Homepage = "https://cheroot.cherrypy.dev"
+"Chat: Matrix" = "https://matrix.to/#/#cherrypy-space:matrix.org"
+"CI: GitHub" = "https://github.com/cherrypy/cheroot/actions"
+"Docs: RTD" = "https://cheroot.cherrypy.dev"
+"GitHub: issues" = "https://github.com/cherrypy/cheroot/issues"
+"GitHub: repo" = "https://github.com/cherrypy/cheroot"
+"Tidelift: funding" = "https://tidelift.com/subscription/pkg/pypi-cheroot?utm_source=pypi-cheroot&utm_medium=referral&utm_campaign=pypi"
+
+[project.readme]
+file = "README.rst"
+content-type = "text/x-rst"
+
+[project.optional-dependencies]
+docs = [
+    # upstream
+    "sphinx>=1.8.2",
+    "jaraco.packaging>=3.2",
+    "sphinx-tabs>=1.1.0",
+
+    # local
+    "furo",
+    # `scm_tag_titles_ext` extension dep in docs/
+    "python-dateutil",
+    "sphinxcontrib-apidoc>=0.3.0",
+]
+
+[project.scripts]
+cheroot = "cheroot.cli:main"
+
+[tool.distutils.bdist_wheel]
```

Comment:
Why is it `distutils` now? I'd expect `setuptools`... Do you have a docs reference to share?

Reply:
Most of this change was programmatically generated, so we'll need to figure out the correct syntax for anything idiosyncratic. I've validated this change against projects in sync with jaraco/skeleton, but since this project is behind, it's liable to stumble on other issues.
I have no idea where `tool.distutils.bdist_wheel` comes from. It was generated by ini2toml. Agreed it looks wrong.

Reply:
This whole section is unneeded for some time. Skeleton dealt with this in jaraco/skeleton@479ac2149d872757160732bc977977ee0192ac51.

Reply:
Dealt with in ef05546d.

Reply:
Is `setuptools` able to read this section from `setup.cfg` when the rest of configuration is in `pyproject.toml`? If yes, I'd just move it back. Although, let's first fix the build env pins to see if it'll work as it is now...

Reply:
> This whole section is unneeded for some time. Skeleton dealt with this in [jaraco/skeleton@479ac21](https://github.com/jaraco/skeleton/commit/479ac2149d872757160732bc977977ee0192ac51).

Are you sure? The skeleton change removes an opposite configuration that used force `py2.py3`. This config used to force `py3`.

Reply:
As you can see, skeleton-based projects haven't been providing this setting for years, and they produce wheels that indicate `py3`. The default behavior is `universal=0`, so being explicit about the default is just debt.

Reply:
I see. Perhaps, at some point the default was different or I wanted to be explicit. Thanks for the explanation!

Reply:
Marking this thread as resolved.